lib/RWMutex.ts

```diff
@@ -65,10 +76,17 @@ export default class RWMutex {
           },
         });
         if (result.matchedCount > 0) {
+          if (typeof cb === "function") {
+            cb();
+          }
           return;
         }
       } catch (err) {
-        throw new Error(`error aquiring lock ${this._lockID}: ${err.message}`);
+        const formattedError = new Error(`error aquiring lock ${this._lockID}: ${err.message}`);
+        if (typeof cb === "function") {
+          cb(formattedError);
```

Comment:
I don't think we want to throw the error though? If you call the function with a callback you get the error through the callback you don't then want an error to be thrown also?